src/nb_type.cpp

```diff
@@ -40,6 +40,72 @@ static PyObject **nb_weaklist_ptr(PyObject *self) {
     return weaklistoffset ? (PyObject **) ((uint8_t *) self + weaklistoffset) : nullptr;
 }
 
+static void nb_enable_try_inc_ref(PyObject *obj) noexcept {
+#if 0 && defined(Py_GIL_DISABLED) && PY_VERSION_HEX >= 0x030E00A5
```

Comment:
Is the 0 intentional here?

Reply:
The APIs haven't been added to CPython yet, but I wanted to get this PR ready while the issue was still fresh in my mind. We can either:

1) Delete this code path for now
2) Wait until the APIs are added
3) Or anything else you prefer

Reply:
I think it's OK as is, we can remove the `#if 0` once the change goes in.

Do I understand correctly that the version below now has an optimization (specific to object construction) that won't be in the official API?

Reply:
I intend to add a similar optimization in the implementation of the CPython API.